postgres.go

```diff
@@ -115,7 +115,24 @@ func (dialector Dialector) DefaultValueOf(field *schema.Field) clause.Expression
 
 func (dialector Dialector) BindVarTo(writer clause.Writer, stmt *gorm.Statement, v interface{}) {
 	writer.WriteByte('$')
-	writer.WriteString(strconv.Itoa(len(stmt.Vars)))
+	index := 0
+	varLen := len(stmt.Vars)
+optionLoop:
+	for varLen > index {
+		switch stmt.Vars[index].(type) {
+		case pgx.QueryResultFormats:
```

Comment:
Can you add a test description at https://github.com/go-gorm/gorm/pull/6806 Is the following necessary?
1. Case `pgx.QueryResultFormats`, `pgx.QueryResultFormatsByOID`, `pgx.QueryRewriter` It seems that these parts are not executed in the test case. In what scenario do we need to handle them?
2. When will the `for varLen > index` loop be used? Currently, I have not found the possibility of `index > 1` from the test cases, which means that the loop is not executed multiple times, it is only executed once.

Reply:
It's not necessary in this scenario   we just need to deal with  one variable of type pgx.QueryExecModeSimpleProtocol , I have removed  pgx.QueryResultFormats, pgx.QueryResultFormatsByOID, pgx.QueryRewriter cases.

Reply:
https://github.com/go-gorm/gorm/pull/6806    gorm can't relay on pgx package directly , so  I can't add unit test  in gorm .

Reply:
> [go-gorm/gorm#6806](https://github.com/go-gorm/gorm/pull/6806) gorm can't relay on pgx package directly , so I can't add unit test in gorm .

Or can you describe the purpose of the `for varLen > index` loop? This seems to be dealing with multiple vars starting with `pgx.QueryExecModeSimpleProtocol`, is this necessary?

Reply:
we'd better to remove  all  vars starting with pgx.QueryExecModeSimpleProtocol,   on matter how many vars we meet.    just to  make the code  more robust . here  for varLen > index   is not necessary  .     I‘ll change  the code as follows

	index := 0
	varLen := len(stmt.Vars)
	if varLen > 0 {
		switch stmt.Vars[index].(type) {
		case pgx.QueryExecMode:
			index++
		}
	}

Reply:
I understand where you're coming from, but I don't think it needs to be dealt with multiple times. The reason is that our purpose of abstracting the driver is to prevent users from caring too much about the details of pgx, that is, we only focus on the relationship between ourselves and pgx (pgx.QueryExecModeSimpleProtocol is only used once). I think the current processing is enough.